### DISCUSSION TITLE
Misorientation distribution

### DISCUSSION THREAD

**Discussion by NovakEvana** [mtex-toolbox/mtex#1679] [Ask Anything] [answered]
Dear MTEX community,

I tried to draw the misorientation distribution chart in mtex; however, it looks quite (much) different from what I retrieved from TSL OIM software (EDAX-based) with the same dataset. I guess the TSL OIM software uses pixel-to-pixel misorientation calculation while it's not the case in mtex. Could you please help to see why I have a difference here? The code was using for mtex is attached below. Thanks!

```
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',15*degree);
grains = smooth(grains,5);
gB = grains.boundary('Martensite_BCC','Martensite_BCC');
figure;
plotAngleDistribution(gB.misorientation)

```
![image](https://user-images.githubusercontent.com/117882416/205517848-b78c201c-8237-451e-adb9-c01c440c5d4f.png)

**Comment by lfgmorales** [comment]
Hi

My educated get is that on the TSL side you used the standard „grain“ parameters, and because of that you have all the low misorientation angles you dont see on yout mtex histogram. Just out of curiosity, why did you choose 15 degrees misorientation?

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 4 Dec 2022, at 23:06, NovakEvana ***@***.***> wrote:

﻿

Dear MTEX community,

I tried to draw the misorientation distribution chart in mtex; however, it looks quite (much) different from what I retrieved from TSL OIM software (EDAX-based) with the same dataset. I guess the TSL OIM software uses pixel-to-pixel misorientation calculation while it's not the case in mtex. Could you please help to see why I have a difference here? The code was using for mtex is attached below. Thanks!

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',15*degree);
grains = smooth(grains,5);
gB = grains.boundary('Martensite_BCC','Martensite_BCC');
figure;
plotAngleDistribution(gB.misorientation)



[image]<https://user-images.githubusercontent.com/117882416/205517848-b78c201c-8237-451e-adb9-c01c440c5d4f.png>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1679>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4LHVVPE563HSBED73WLUIUPANCNFSM6AAAAAASTTGKTQ>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1679", "url": "https://github.com/mtex-toolbox/mtex/discussions/1679", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by NovakEvana** [reply]
Hi Luiz,

Thanks for your answer. 
`My educated get is that on the TSL side you used the standard „grain“ parameters, and because of that you have all the low misorientation angles`. Can you tell me how to plot in mtex using that "grain" parameter too?

`Just out of curiosity, why did you choose 15 degrees misorientation?` I followed a rule of thumb from conventionally-produced steels (see below image) where we separate LAGB and HAGB by their energy at the grain boundary. I know that is a poor assumption but the grain size follows the same trend in a relative comparison regardless of the chosen misorientation. 

![image](https://user-images.githubusercontent.com/117882416/205585021-93d9097a-0b8a-442b-acd4-839d2b7c8bfd.png)

**Reply by NovakEvana** [reply]
Hi Luiz and community,

Any idea on this? Thanks.

**Comment by lfgmorales** [comment]
Hi

You can either decrease your misorientation threshold, or better, compute two thresholds, something like

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[1*degree, 10*degree])

You may get some more information here
<https://mtex-toolbox.github.io/SubGrainBoundaries.html>
Subgrain Boundaries | MTEX<https://mtex-toolbox.github.io/SubGrainBoundaries.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/SubGrainBoundaries.html>
        [favicon.png] <https://mtex-toolbox.github.io/SubGrainBoundaries.html>



Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 6 Dec 2022, at 17:08, NovakEvana ***@***.***> wrote:

﻿

Hi Luiz and community,

Any idea on this? Thanks.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1679#discussioncomment-4324657>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7ZE2FHNGXIG7KFTGTWL5QFFANCNFSM6AAAAAASTTGKTQ>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1679#discussioncomment-4324657", "url": "https://github.com/mtex-toolbox/mtex/discussions/1679#discussioncomment-4324657", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by ralfHielscher** [comment, marked as answer]
Hi Evana,

The difference is in the angle distribution is indeed very extreme. Could you please check the following:

1. In order to get an comparable histogram you should segment at 5 degree threshold angle, i.e. 
  `[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree)`
2.  Is the grain reconstruction similar in TSL and MTEX?
3.  You can highlight the 5-50° boundaries in TSL and in MTEX (by `plot(gB(gB.misorientation.angle<50*degree,'linecolor','r','linewidth',2)`. You should find many grain boundaries in TSL and only few in MTEX. Which are those. Inspect the neighboring grain orientations. Only one can be right.

Ralf.

**Comment by NovakEvana** [comment]


**Reply by kilir** [reply]
In addition, does TSl consider any boundary? As of now, you are only considering closed boundaries in Mtex.
```
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[5*degree, 15*degree])
b = [grains.boundary('M','M') grains('M').innerBoundary]
```
Cheers,
Rüdiger

**Reply by NovakEvana** [reply]
Thanks Rudiger, TSL also considers inner boundaries. And it works too. Thank you so much,
